Add copies of the main js files for code review

peterh32 opened the pull request:
Do not merge this branch! For code review only!!

leondragan commented:
BoxButton - I would avoid having li as the containing element as that button can now only exist inside UL element.  Solutions might be call the item as DraggableListItem which is more semantically accurate or taking hte LI out of the component

BoxOButtons - 

make sure that function parameter names are descriptionve.  setTarget(n) does not tell me what n is supposed to be.

leondragan commented:
BoxOButtons ``render()``
Why are we inserting _anything_ into the head element??? Components should be self contained.

why are setting stuff on invisible as opposed to allowing the nodes to be dropped and recreated?  If there is legitimate reason, please provide comments.

leondragan commented:
Why are we using gulp instead of webpack?

leondragan commented:
BoxOButtons ``generateDropAcceptedEvent()``
user ``let`` instead of ``var`` in es6

peterh32 commented:
Thanks for the comments @leondragan .
> BoxButton - I would avoid having li as the containing element...
Good idea. Changed to div.

> setTarget(n) does not tell me what n is ...
I changed n to buttonIndex. That's much more descriptive.

> Why are we inserting anything into the head element?
That's a hack to insert the css. We can do it some other way in production, but for this demo page, that was the simplest approach to make the thing self-contained.

> why are setting stuff on invisible as opposed to allowing the nodes to be dropped and recreated?
That is a good question! 
Architectural answer: The highlight element is the blue line that indicates the current highlight state of the element. It's tightly coupled to the box, basically like an outline that you can turn on and off with CSS. Also it has no interactions of its own. For those reasons, I did not think it belonged in its own component.
Functional answer: The highlight element takes up space. If you remove and re-add it, you'll mess with the wrapping of buttons in the box, and cause strange jumping behaviors as the user drags elements around. For that reason it's best to leave the element in place and simply toggle its visibility.

> Why are we using gulp instead of webpack?
That is just a way of getting this component repository to work. When this gets integrated into our actual app, we'll use whatever the PP app uses.

> use let instead of var in es6
Got it, thanks!
